subprojects/android-test/snackbar-rule/src/main/kotlin/com/avito/android/test/app/second/SnackbarProxyRule.kt

```diff
@@ -0,0 +1,62 @@
+package com.avito.android.test.app.second
+
+import android.annotation.SuppressLint
+import android.widget.TextView
+import com.avito.android.rule.SimpleRule
+import com.avito.android.snackbar.proxy.SnackbarProxy
+import com.avito.android.snackbar.proxy.SnackbarProxyHolder
+import com.avito.android.test.waitFor
+import com.google.android.material.snackbar.Snackbar
+import org.junit.Assert
+import com.google.android.material.R as google_R
+
+class SnackbarProxyRule : SimpleRule() {
+
+    private val proxy = Proxy()
+
+    override fun before() {
+        SnackbarProxyHolder.proxy = proxy
+    }
+
+    fun isShown(text: String) {
+        waitFor {
+            Assert.assertTrue("Snackbar with text=\"$text\" wasn't shown", proxy.snackbarTexts.any { it == text })
+        }
+    }
+
+    fun isShownLast(text: String) {
+        waitFor {
+            Assert.assertTrue(
+                "Snackbar with text=\"$text\" wasn't shown last",
+                text == proxy.snackbarTexts.lastOrNull { it == text })
+        }
+    }
+
+    fun isShownTimes(text: String, times: Int) {
+        waitFor {
+            Assert.assertTrue(
+                "Snackbar with text=\"$text\" wasn't shown $times times",
+                times == proxy.snackbarTexts.count { it == text })
+        }
+    }
+
+    fun clear() {
+        proxy.snackbarTexts.clear()
+    }
+
+    @SuppressLint("VisibleForTests")
+    private class Proxy : SnackbarProxy {
+        val snackbarTexts = mutableListOf<String>()
```

Comment:
I decide to make the model simple. And do not save the whole Snackbar.
If we think I should save Snackbar or create a model to replace Sting we could discuss it
But I consider that code simple and straight forward

Reply:
Snackbar also has an action button to tap on and it's text to check
Should we also support it right in this PR to be sure we don't like basic functionality?

Reply:
I don't know if that check makes sense I could add the possibility for checking the action button text.